osu.Game.Rulesets.Mania/Skinning/Argon/ManiaArgonSkinTransformer.cs

```diff
@@ -92,54 +103,208 @@ public ManiaArgonSkinTransformer(ISkin skin, IBeatmap beatmap)
 
                     case LegacyManiaSkinConfigurationLookups.ColumnBackgroundColour:
 
-                        Color4 colour;
+                        var colour = getColourForLayout(column, stage);
 
-                        const int total_colours = 7;
+                        return SkinUtils.As<TValue>(new Bindable<Color4>(colour));
+                }
+            }
 
-                        if (stage.IsSpecialColumn(column))
-                            colour = new Color4(159, 101, 255, 255);
-                        else
-                        {
-                            switch (column % total_colours)
-                            {
-                                case 0:
-                                    colour = new Color4(240, 216, 0, 255);
-                                    break;
+            return base.GetConfig<TLookup, TValue>(lookup);
+        }
 
-                                case 1:
-                                    colour = new Color4(240, 101, 0, 255);
-                                    break;
+        private Color4 getColourForLayout(int column, StageDefinition stage)
+        {
+            // For now, these are defined per column count as per https://user-images.githubusercontent.com/50823728/218038463-b450f46c-ef21-4551-b133-f866be59970c.png
+            // See https://github.com/ppy/osu/discussions/21996 for discussion.
+            switch (stage.Columns)
+            {
+                case 1:
+                    return colour_yellow;
 
-                                case 2:
-                                    colour = new Color4(240, 0, 130, 255);
-                                    break;
+                case 2:
+                    switch (column)
+                    {
+                        case 0: return colour_green;
 
-                                case 3:
-                                    colour = new Color4(192, 0, 240, 255);
-                                    break;
+                        case 1: return colour_cyan;
 
-                                case 4:
-                                    colour = new Color4(0, 96, 240, 255);
-                                    break;
+                        default: throw new ArgumentOutOfRangeException();
```

Comment:
As caught by CI, this doesn't work with the dual stages mod. The switch should be over `column % 2, 3, 4...` in each of the column count cases here.